content/technotes/2020-02-26-tic-ghactions.Rmd

```diff
@@ -0,0 +1,135 @@
+If packages are updated on CRAN, {tic} would update those before proceeding to the "script" stage.
+After some time all packages would be outdated and the cache quite useless.
+Hence, {tic} rebuilds the cache daily.
+
+The cache rebuilding happens in a scheduled CRON run daily at 3 am in the morning.
+Because {tic} always uploads the cache even if the build failed ([r-lib/actions]() only uploads on a successful build), there will always be a cache available for builds which start during the day.
+
+So even if the build of a repository takes several minutes to install all dependencies, the first commit every day will always make use of an existing cache.
+
+### Installation of dependencies
+
+{tic} uses `ccache` on macOS and Linux to speed up R package installations.
+This tool helps speeding up the installation of compiled packages by loading the C code parts from the cache.
+{tic} rebuilds the `ccache` only **once a week** making it possible that the CRON builds in the morning can make use of the `ccache` builds when rebuilding the R package cache.
+
+### macOS compiler toolchain
+
+macOS is a complicated platform for R.
+CRAN does use a custom compiler for creating the binaries for this platform.
+One reason is that the default `/usr/bin/clang` does not support `openMP`.
+For the current R 3.6.x release a custom version of `clang7` is used while for R 4.x `clang8` will be used.
+{tic} reflects exactly this setting on the macOS-release and macOS-devel runners, eliminating lots of issues when compiling packages from source and giving you the security to have the CRAN setup mirrored in your builds.
+
+### Addtional small tweaks
+
+- {tic} always runs `R CMD javareconf` to ensure a working installation of {rJava}
+- {tic} uses four cores for installing dependencies instead of just running sequentially
+- {tic} sets the env var `RGL_USE_NULL` to make it possible to install the {rgl} package if needed
+- {tic} installs all packages from source on Linux instead of using the R package binaries provided by RStudio.
+  This is to mimic the normal behavior on Linux and ensure a proper linking against system libraries.
+
+## Getting Started
+
+If you are triggered now to try out {tic} and GitHub Actions, here is how to get started:
+
+```{r, eval=FALSE}
+library("tic")
+
+use_ghactions_deploy() # (optional) setup deployment from builds
+use_ghactions_yml() # use deploy = TRUE to deploy from builds
+use_tic_r() # add tic.R (replace `ci_on_travis()` by `ci_on_ghactions()`)
+use_tic_badge("ghactions") # (optional) add a badge to README.md/README.Rmd
+```
+
+Alternatively you can also run `tic::use_tic()` to start an interactive wizard guiding you through the setup process.
+
+After that, commit `.github/workflows/main.yml` and `tic.R` and have a look at the "Actions" pane in the repository.
+
+## Summary
+
+{tic} aims to simplify CI related tasks for R projects (not just packages by the way) and now provides supercharged support for GitHub Actions.
```

Comment:
In the conclusion re-acknowledge that the GH Actions support is built on top of r-lib/actions?